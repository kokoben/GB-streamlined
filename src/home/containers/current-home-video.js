import React, { Component } from 'react';
import CurrentVideo from '../../components/current-video';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { setVideo } from '../actions';
import { apiKey } from '../../api-keys';

class CurrentHomeVideo extends Component {
  componentDidMount() {
    this.props.setVideo();
    console.log(this.props.latestVideo);
  }

  render() {
    if (this.props.latestVideo == null) return null;

    return (
        <CurrentVideo 
          name={this.props.latestVideo.name}
          poster={this.props.latestVideo.image.screen_large_url}
          user={this.props.latestVideo.user}
          pub_date={this.props.latestVideo.publish_date}
          description={this.props.latestVideo.deck}
          source={this.props.latestVideo.hd_url + '?api_key=' + apiKey }
          />
    )
  }
}

const mapStateToProps = (state) => {
  return {
    latestVideo: state.latestVideo
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    setVideo: setVideo
  }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(CurrentHomeVideo);

