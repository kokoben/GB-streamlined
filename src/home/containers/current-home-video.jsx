import React, { Component } from 'react';
import CurrentVideo from '../../components/current-video';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { setLatestHomeVideo } from '../actions';

class CurrentHomeVideo extends Component {
  componentDidMount() {
    this.props.setLatestHomeVideo();
  }

  render() {
    if (this.props.homeVideo === null) return null;

    return (
      <CurrentVideo 
        name={this.props.homeVideo.name}
        user={this.props.homeVideo.user}
        embed_player={this.props.homeVideo.embed_player}
        pub_date={this.props.homeVideo.publish_date}
        deck={this.props.homeVideo.deck}
        />
    )
  }
}

const mapStateToProps = state => ({
  homeVideo: state.homeVideo,
  latestHomeVideoId: state.latestHomeVideoId
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setLatestHomeVideo,
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(CurrentHomeVideo);

