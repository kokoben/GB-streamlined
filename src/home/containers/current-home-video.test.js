import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CurrentHomeVideo } from './current-home-video';

it('renders correctly', () => {
  const mockSetLatestHomeVideo = jest.fn();
  const homeVideo = {
    name: "name",
    user: "user",
    embed_player: "embed_player",
    publish_date: "pub_date",
    deck: "deck",
  }

  const wrapper = shallow(<CurrentHomeVideo
    homeVideo={homeVideo}
    setLatestHomeVideo={mockSetLatestHomeVideo}
  />);

  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
