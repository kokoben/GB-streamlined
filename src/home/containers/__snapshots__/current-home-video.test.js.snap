// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<CurrentVideo
  deck="deck"
  embed_player="embed_player"
  name="name"
  pub_date="pub_date"
  user="user"
/>
`;
